From: Daniel DeGrasse <daniel@degrasse.com>
Date: Mon, 12 Aug 2024 08:41:38 -0700
Subject: [PATCH] datetime: fix disabling of auto timezone during panel load

Commit 021fce8c9 (datetime: Don't show Auto Timezone setting
without location services) added code that unconditionally disables
the "automatic-timezone" dconf key while loading the datetime
panel view. Make this code conditional, so it only executes if
HAVE_LOCATION_SERVICES is not defined by the compiler.

Fixes #3157

Signed-off-by: Daniel DeGrasse <daniel@degrasse.com>
(cherry picked from commit 213650b521073fade14410d57fae9f3d46cadbf9)

Origin: https://gitlab.gnome.org/GNOME/gnome-control-center/-/merge_requests/2785
Bug-Ubuntu: https://launchpad.net/bugs/2073430
Bug-GNOME: https://gitlab.gnome.org/GNOME/gnome-control-center/issues/3157
Last-Update: 2024-09-11
---
 panels/system/datetime/cc-datetime-page.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/panels/system/datetime/cc-datetime-page.c b/panels/system/datetime/cc-datetime-page.c
index 17bb32a..c057c68 100644
--- a/panels/system/datetime/cc-datetime-page.c
+++ b/panels/system/datetime/cc-datetime-page.c
@@ -926,8 +926,6 @@ cc_date_time_page_init (CcDateTimePage *self)
   self->auto_timezone_supported = FALSE;
 
   self->datetime_settings = g_settings_new (DATETIME_SCHEMA);
-  // Let's disable location services for gnome-46
-  g_settings_set_boolean (self->datetime_settings, AUTO_TIMEZONE_KEY, FALSE);
 #ifdef HAVE_LOCATION_SERVICES
   self->auto_timezone_supported = TRUE;
   gtk_widget_set_visible (GTK_WIDGET (self->auto_timezone_row), TRUE);
@@ -941,6 +939,9 @@ cc_date_time_page_init (CcDateTimePage *self)
   g_settings_bind (self->datetime_settings, AUTO_TIMEZONE_KEY,
                    self->auto_timezone_row, "active",
                    G_SETTINGS_BIND_DEFAULT);
+#else
+  // Let's disable location services
+  g_settings_set_boolean (self->datetime_settings, AUTO_TIMEZONE_KEY, FALSE);
 #endif
 
   /* Clock settings */
